### DISCUSSION TITLE
Pull workflow.revision into config scopes

### DISCUSSION THREAD

**Discussion by tjdurant** [nextflow-io/nextflow#3130] [Q&A] [answered]
Hi Nextflow Community,

We really like the Nextflow functionality that allows us to run our pipeline with commands like:

```
nextflow run <git_repo_url> -r <tag>
```

The way we have our configuration file set up is that it specifies the output path for files that are generated and as part of that path, we include the pipeline version - i.e., `~/PIPELINE_OUTPUT_v1.3.0/path/to/files`

We currently get the pipeline version into the config by specifying it as a parameter and declaring it at runtime. But, then we have to specify that parameter in two places if we want to use `-r` and our parameter, `--pipeline_version` which creates a scenario where the two variables could be erroneously different. 

One option is to check for this discrepancy as described [here](https://github.com/nextflow-io/nextflow/issues/818#:~:text=if%20(%20params.one%20%26%26%20params.two%20)%20%7B%20error%20%22Params%20one%20and%20two%20collide%22%20%7D). And just specify the pipeline_version/tag in multiple places in the runtime command. Althought it would be nice to specify in just one place. 

So, currently, we have a bash script that takes the pipeline_version/tag and sticks it in all the places we want it for the runtime command. Although, this is a little suboptimal because the person running it needs to go to the repo directory and `git checkout` to the stable tag to make sure they have the appropriate bash script and then run it. 

We would really like it if in our config we could pull in `workflow.revision` and include it as a piece of that path that we define in our config file. For example:

```
params {
    results = "~/PIPELINE_OUTPUT_${worflow.revision}/path/to/files"
}

report {
    file = "${params.results}/pipeline_report"
}
```

Is there a way to do this that we are missing? It seems that when we put any variables within a scope in config, the scope is appended to the front of that variable (e.g., params.workflow.revision) and throws an error. 

We discussed the possibility of defining the 'results' path in the main.nf script but don't know how to handle the `report.file` definition if we were to go that route, because as far as we know, we can't define that path in the main.nf script. 

Any thoughts/comments/suggestions would be greatly appreciated!

Thanks
T

**Comment by pditommaso** [comment, marked as answer]
Think at this time the best practice is to use a param for that. You may want to open an issue to track this is a feature request.

**Reply by tjdurant** [reply]
@pditommaso thanks! Just did [here](https://github.com/nextflow-io/nextflow/issues/3133)
